autodispose/src/test/java/com/uber/autodispose/AutoDisposeObserverTest.java

```diff
@@ -171,6 +177,24 @@
     assertThat(lifecycle.hasObservers()).isFalse();
   }
 
+  @Test public void autoDispose_withScopeProviderCompleted_shouldNotReportDoubleSubscriptions() {
+    RxJavaPlugins.setErrorHandler(new Consumer<Throwable>() {
+      @Override
+      public void accept(Throwable throwable) throws Exception {
+        assertThat(throwable instanceof ProtocolViolationException).isFalse();
```

Comment:
This test case fails in master as:
`Exception in thread "main" io.reactivex.exceptions.ProtocolViolationException: It is not allowed to subscribe with a(n) com.uber.autodispose.AutoDisposingObserverImpl multiple times. Please create a fresh instance of com.uber.autodispose.AutoDisposingObserverImpl and subscribe that to the target source instead.
	at com.uber.autodispose.AutoDisposeEndConsumerHelper.reportDoubleSubscription(AutoDisposeEndConsumerHelper.java:110)
	at com.uber.autodispose.AutoDisposeEndConsumerHelper.setOnce(AutoDisposeEndConsumerHelper.java:56)
	at com.uber.autodispose.AutoDisposingObserverImpl.onSubscribe(AutoDisposingObserverImpl.java:67)
	at io.reactivex.subjects.PublishSubject.subscribeActual(PublishSubject.java:85)
	at io.reactivex.Observable.subscribe(Observable.java:10842)
	at com.uber.autodispose.ObservableScoper$AutoDisposeObservable.subscribeActual(ObservableScoper.java:113)
	at io.reactivex.Observable.subscribe(Observable.java:10842)
	at com.uber.autodispose.ObservableScoper$1.subscribe(ObservableScoper.java:94)
	at com.uber.autodispose.AutoDisposeObserverTest.autoDispose_withScopeProviderCompleted_shouldNotReportDoubleSubscriptions(AutoDisposeObserverTest.java:192)`

Reply:
I don't quite understand this test, as the `ProtocolViolationException` isn't obvious. Some sort of comment and asserting what you do expect should happen here

It also won't fail if this callback isn't hit, let's make it explicit somehow that we expect this callback to be hit